### PR TITLE
Fix empty time durations

### PR DESCRIPTION
**Pull request by uklotzde**
Qt5: If QTime is null it stays null even if you add something to it.

When searching for "QTime()" I found some other minor issues that I fixed.

### CODE REVIEW

**Comment by Pegasus-RPG** [comment]
You ran the tests with this change built against Qt4.x too, right?

**Comment by uklotzde** [comment]
Not yet. But I hope the build server does it for me ;) A local Qt5 build is 
still running.

I pushed some additional small fixes to clean up the code that uses 
time.h/.cpp. Now the plugin build does no longer depend on 
Time::formatSeconds() to be inlined.

**Comment by rryan** [comment]
> Not yet. But I hope the build server does it for me ;) A local Qt5 build is
> still running.

Yea, Travis being green is good enough to tell us it's fine.

File: src/util/time.cpp [hunk 1]
```diff
@@ -6,3 +6,33 @@ LLTIMER Time::s_timer;
...
+
+    // NOTE(uklotzde): Time() constructs a 'null' object, but
+    // we need 'zero' here.
+    QTime t = QTime(0, 0).addMSecs(dSeconds * kMillisPerSecond);
+
+    QString formatString =
+            (days > 0 ? (QString::number(days) %
+                         QLatin1String("'d', ")) : QString()) %
+            QLatin1String(days > 0 || t.hour() > 0 ? "hh:mm:ss" : "mm:ss") %
+            QLatin1String(showCentis ? ".zzz" : "");
```
Comment (daschuer) [inline src/util/time.cpp]: Unrelated, but I have just noticed it: 
http://stackoverflow.com/questions/31477309/timestamp-in-iso-8601-the-last-6-digits-yyyy-mm-ddthhmmss
it should be QLocale::decimalPoint()

**Comment by daschuer** [comment]
LGTM.  If you think we should fix the decimal point issue later just give a hint.

File: src/util/time.cpp (hunk 1 above)
Comment (uklotzde) [inline src/util/time.cpp]: QLocale::decimalPoint() is a separate issue.

But I added a unit test and it turns out that the formatting algorithm is 
wrong! Fix and tests will follow.

**Comment by rryan** [comment]
I would suggest fixing a separate issue in a separate PR :). Travis says :+1: so I say LGTM :)

**Comment by uklotzde** [comment]
I've just added the formatting fix and the unit tests. Bug: The micro/centi 
seconds were missing in some cases.

Minor issue: I've also removed the strayed single quotes that where placed 
around days postfix "d".

**Comment by uklotzde** [comment]
You were right, RJ! We better fix those inconsistencies in a separate task. 
I've reverted my last commit immediately.

Strangely the micro/centi seconds are only added for large durations >= 1 
hour. But the library view looks strange with microsecond durations on 
single tracks. Looks like we need more formatting options.

**Comment by uklotzde** [comment]
Finished. The formatting fixes will follow in a separate PR.

**Comment by daschuer** [comment]
Thank you very much. A nice bug, hard to spot.
LGTM!

**Pull request merged**